kedro/extras/datasets/geojson/geojson_dataset.py

```diff
@@ -0,0 +1,163 @@
+allowed geopandas (pandas) options for loading and saving geosjon files.
+"""
+import copy
+from pathlib import PurePosixPath
+from typing import Any, Dict, Union
+
+import fsspec
+import geopandas as gpd
+
+from kedro.io.core import (
+    AbstractVersionedDataSet,
+    DataSetError,
+    Version,
+    get_filepath_str,
+    get_protocol_and_path,
+)
+
+
+class GeoJSONDataSet(AbstractVersionedDataSet):
+    """``GeoJSONDataSet`` loads and saves data to a geojson file using an underlying filesystem
+    (eg: local, S3, GCS).
+    The underlying functionality is supported by geopandas, so it supports all
+    allowed geopandas (pandas) options for loading and saving geosjon files.
+
+    Example:
+    ::
+
+        >>> import geopandas as gpd
+        >>> from shapely.geometry import Point
+        >>> from kedro.io.contrib import GeoJSONDataSet
+        >>> data = gpd.GeoDataFrame({'col1': [1, 2], 'col2': [4, 5],
+        >>>                      'col3': [5, 6]}, geometry=[Point(1,1), Point(2,4)])
+        >>> # data_set = GeoJSONDataSet(filepath="gcs://bucket/test.geojson",
+        >>>                                save_args=None)
+        >>> data_set = GeoJSONDataSet(filepath="test.geojson",
+        >>>                                save_args=None)
+        >>> data_set.save(data)
+        >>> reloaded = data_set.load()
+        >>>
+        >>> assert data.equals(reloaded)
+
+    """
+
+    DEFAULT_LOAD_ARGS = {}  # type: Dict[str, Any]
+    DEFAULT_SAVE_ARGS = {"driver": "GeoJSON"}  # type: Dict[str, Any]
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        filepath: str,
+        load_args: Dict[str, Any] = None,
+        save_args: Dict[str, Any] = None,
+        version: Version = None,
+        credentials: Dict[str, Any] = None,
+        fs_args: Dict[str, Any] = None,
+    ) -> None:
+        """Creates a new instance of ``GeoJSONDataSet`` pointing to a fsspec
+        filepath.
+
+        Args:
+
+            filepath: path to a geojson file. Should be a ``fsspec```
+                protocol path
+            load_args: GeoPandas options for loading GeoJSON files.
+                Here you can find all available arguments:
+            save_args: GeoPandas options for saving geojson files.
+                Here you can find all available arguments:
+                The default_save_arg driver is 'GeoJSON', all others preserved.
+            version: If specified, should be an instance of
+                ``kedro.io.core.Version``. If its ``load`` attribute is
+                None, the latest version will be loaded. If its ``save``
+            credentials: credentials required to access the underlying filesystem.
+                Eg. for ``GCFileSystem`` it would look like `{'token': None}`.
+            fs_args: Extra args to pass into the underlying filesystem
+                Eg. for ``GCFileSystem`` it would look like `{'project': 'my-project' , ...}`.
```

Comment:
Can you please look at `json_dataset.py` and make the docstring more less consistent with it?